service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -130,6 +130,15 @@ public String getUserEmailFromSam(AuthenticatedUserRequest userRequest)
     }
   }
 
+  public String getProxyGroupEmail(String userEmail) throws InterruptedException {
+    GoogleApi googleApi = samGoogleApi(getWsmServiceAccountToken());
```

Comment:
I think this method should also take in an `AuthenticatedUserRequest` and use those credentials here, instead of pulling WSM's SA credentials. 
WSM's SA has broad permissions and it makes determining the real caller from logs harder, so we generally avoid using it unless we need to.